perfkitbenchmarker/providers/openstack/os_virtual_machine.py

```diff
@@ -120,18 +143,22 @@ def _PostCreate(self):
             instance = self.client.servers.get(self.id)
             status = instance.status
 
-        with self._floating_ip_lock:
-            self.floating_ip = self.public_network.get_or_create()
-            instance.add_floating_ip(self.floating_ip)
-            logging.info('floating-ip associated: {}'.format(
-                self.floating_ip.ip))
+        assert self.private_net is not None  # Previously checked, must be true.
```

Comment:
I recommend adding a string to this assert, or at least modifying the comment to say that the previous check was in _Create.